net/nimble/host/include/host/ble_gap.h

```diff
@@ -98,7 +98,7 @@ struct hci_conn_update;
 
 #define BLE_GAP_EVENT_CONNECT               0
 #define BLE_GAP_EVENT_DISCONNECT            1
-#define BLE_GAP_EVENT_CONN_CANCEL           2
+#define BLE_GAP_EVENT_CONN_CANCEL           2 /* XXX: Deprecated. */
```

Comment:
We should remove this symbol as well (or rename it to reserved as discussed) since code which uses this event will complie, but won't receive this event anymore which is confusing. Also corresponding structure form GAP event structure should be removed.

Another option is to keep this for one release but we should send both old and new events to retain compatibility.
I'd go with removal and we'll put note about this in release notes. It's not a big change and won't require major changes in existing apps.